Clown Tech Overhaul: Social Psychology and Unregulated Pranking

alphanerdd opened the pull request:
## About The Pull Request
I was inspired by [this post](https://tgstation13.org/phpBB/viewtopic.php?p=622102#p622102) in the forums that points out the shortcomings of Clown tech. This PR aims to rebrand Clown Tech to Social Psychology, expand what can be printed from the service protolathes, and add some fun new illegal tech synergies.

Clown tech is replaced with the **Social Psychology** node, which in addition to all the things old clown tech did, now includes recipes for dice bags, crayon boxes, cardboard cutouts, synthesizers, and decks of cards. All of these are available from the service protolathe.

With both Social Psychology and Illegal tech researched, a new node, **Unregulated Pranking** is unlocked. This enables adaptive cardboard cutouts, pneumatic pie cannons, and decoy nuclear authentication disks to be printed from the service protolathe. 
![unregulated_pranking](https://user-images.githubusercontent.com/60521518/152231797-d038bf10-09cb-4922-8898-47a1f473eba9.png)
## Why It's Good For The Game
Previously, clown tech only unlocked clown borgs, clown mechs, and airhorns. That's both useless and disappointing in my opinion. Now with Social Psychology, these clown items are tied to multiple civilian items that really should be printable via the lathe.  (also adding more nonessential items that require plastic may encourage chemists to make it occasionally.)

Unregulated pranking makes a few _mostly_ harmless syndicate items available to the public. The tech already has quite a high barrier to entry, considering all the perquisites for illegal tech, and in return provides items that can be mad funny. Cardboard cutout and decoy disk gimmicks are pretty funny, and it would be cool to see more of those in the future.

As for the other suggestions in the forum post, I rarely use the art tools in game, so I'm probably not the best person to make those changes. I am however a dedicated mime fan, and would love to see new mime items. However ~~I'm not creative enough to make new mime stuff~~ i wanted to atomize that into a separate PR at some point.
## Changelog
:cl:
del: clown tech node
add: social psychology node, which unlocks dice, cards, crayons, cutouts, synthesizers, and bike horns in the service protolathe
add: unregulated pranking node, which unlocks adaptive cutouts, decoy disks, clown PDA cartridges, and pie cannons in the service protolathe
tweak: cardboard can be accepted by lathes and ore silos now
/:cl

alphanerdd committed:
Rename Social Psych

changes clown tech to social psychology.

alphanerdd committed:
Add dice bag to protolathe

adds the dice bag to the protolathe as well as the social psych tech. future items will be added in bulk, testing for now

alphanerdd committed:
comma lol

lol

alphanerdd committed:
spelling

me code good

alphanerdd committed:
Cards and Crayons

adds cards and crayon boxes to the protolathe

alphanerdd committed:
Add synthesizer

adds synthesizer to protolathe. better than adding every instrument imo.

alphanerdd committed:
Unregulated Pranking

adds the unregulated pranking node, requires illegal tech. only includes adaptive cardboard cutouts rn

alphanerdd committed:
Modifying for debug

shitcode lmao

alphanerdd committed:
re-hides advanced pranks, adds normal cutouts to psych

pretty sure the thing is fixed, also adding normal cutouts

alphanerdd committed:
adds decoy disk

decoy disk is now printable

alphanerdd committed:
Fixes runtime, updates prereqs

Did you know that Entertainment Modular Suits have clown tech as a prerequisite? The wiki doesn't.

alphanerdd committed:
testing

alphanerdd committed:
Revert "testing"

This reverts commit 1d07c5ac85b1e865e011670489ace849d6084593.

alphanerdd committed:
Adds the pie cannon

pie cannon

alphanerdd committed:
shitcode lol lmao

Bawhoppen commented:
Having producible nuke disk decoys is a bit concerning to me. Perhaps it could be a BEPIS-activated node, so that way it's uncommon but a fun thing for an ambitious clown to strive for?

alphanerdd commented:
> Having producible nuke disk decoys is a bit concerning to me. Perhaps it could be a BEPIS-activated node, so that way it's uncommon but a fun thing for an ambitious clown to strive for?

With all the requirements for illegal tech you shouldn't really be able to unlock this until fairly late into the shift, and not reliably since you need syndie gear (and for traitors the extra effort to save 1 tc on a disk isnt really worth it imo). At that point either the captain has the disk on them or nukies are here and half the station is exploded. Still making the resource costs a bit steeper, or having it require even rarer resources, might be a good idea so the halls aren't flooded with them.

ATH1909 commented on code/modules/research/designs/weapon_designs.dm:
```diff
@@ -362,3 +362,13 @@
+	desc = "Load cream pie for optimal results."
+	id = "pie_cannon"
+	build_type = PROTOLATHE | AWAY_LATHE
+	materials = list(/datum/material/iron = 10000, /datum/material/bananium = 3000)
```
requires bananium, sad

alphanerdd replied:
Clowns stay mad, they have to work for a second pie cannon

jughu commented on code/modules/research/designs/misc_designs.dm:
```diff
@@ -334,6 +334,76 @@
+	id = "crayon_box"
+	build_type = PROTOLATHE | AWAY_LATHE
+	materials = list(/datum/material/plastic = 1000)
+	build_path = /obj/item/storage/crayons
```
Why would plastic get you crayons 🤨

jughu replied:
(make it require colourfull reagent i know you can put chemicals in the techlathe)

jughu replied:
yeah thats far, only instance of chems used i know is the flomatray spray(botanist plant mutator gun) thought colourfull agent could just make the crayons for you like it isn't that hard of recipe to make you could also just make it require a low amount of it like 7 colourfull reagent for 1 box 1 colourfull reagent unit for each crayon

alphanerdd replied:
sorry for the late response, i was a bit busy over the weekend. As far as I'm aware there aren't any other recipes in the protolathe the require chems currently. From what I've heard it used to be used for circuit printing, but not anymore. To be honest I agree with the decision to not require chems for lathe recipes, since it makes crafting the item a bit unintuitive and complicated. Regardless, I don't think having only one tech locked recipe require chems is a good idea. If using chems with protolathes is going to be a thing, it probably needs to be more prevalent. Since I added cardboard as a lathe mat, I decided to replace half of the recipe with cardboard, since your also printing a box. Cuts the plastic cost down a bit and adds another recipe requiring cardboard, which is good imo.

jughu commented on code/modules/research/designs/misc_designs.dm:
```diff
@@ -334,6 +334,76 @@
+	materials = list(/datum/material/plastic = 2000)
+	build_path = /obj/item/cardboard_cutout
+	category = list("Equipment")
+	departmental_flags = DEPARTMENTAL_FLAG_SERVICE
```
Plastic doesn't make sense for cardboard cutouts either maybe make the techlathe accept cardboard as well(making it an acceptable mat datum)

alphanerdd replied:
Yeah i was thinking something like synthetic cardboard, mainly to explain how it can be printed from a lathe. To be honest i wouldn't be against having cardboard usable in lathes, but is it okay to have a lathe material that's only used in two recipes?

alphanerdd replied:
it hath been done

timothymtorres commented:
This PR is missing the biggest thing that makes a clown HONK.  The bikehorn...  Sad.

Sealed101 commented:
add the clown pda cartridge as well so people can print them en masse and upload the honk virus to anything and everything
also imagine calling the honk mech useless smh

alphanerdd commented:
> This PR is missing the biggest thing that makes a clown HONK. The bikehorn... Sad.

I have made a grave error, this will be fixed immediately

alphanerdd commented:
> add the clown pda cartridge as well so people can print them en masse and upload the honk virus to anything and everything also imagine calling the honk mech useless smh

Oh yeah clown pda cartridge is a great idea. It should probably be locked behind unregulated pranking tho, since infinite honk viruses is a terrifying concept.

Honk Mech is low tier, change my mind

alphanerdd committed:
add bike horn to social psych

i'm not sure how i forgot this. I also re-arranged some things in misc_designs, sectioning off all of the social psych recipes.

alphanerdd committed:
add clown cartridge to unregulated pranks

catastrophic levels of clowning. made it require a little bit of bananium ~~because i hate clowns~~ so it cant be spammed as much

alphanerdd committed:
awful job, turboshit

im bad at this

me dumb, hit equals key twice

alphanerdd committed:
makes cardboard an acceptable mat datum

makes cardboard acceptable, used in place of plastic with cutouts. Probably a good idea to add more recipes for this in the future

Sealed101 commented:
i will not change your mind just so we dont get more dumbfuck shifts where a major threat was honkerblastered and glovepunched into non existence thank you

alphanerdd committed:
changes crayon box to include cardboard

lowers plastic requirement

alphanerdd committed:
Merge branch 'tgstation:master' into master

alphanerdd committed:
Revert "Merge branch 'tgstation:master' into master"

This reverts commit 661b7f4842ad367bdaf2c386eb833b68d658fd29, reversing
changes made to 005441626262f602c69075aa08d78c4a172d9e30.

alphanerdd commented:
major fuckup hold on

alphanerdd committed:
Revert "Revert "Merge branch 'tgstation:master' into master""

This reverts commit 7732b065ad58d264ff025291d1e43a761c84d0a0.

alphanerdd commented:
> major fuckup hold on

reverted the wrong thing, i am a dumb fucking idiot. pretty sure it's fixed now, but it requested review from like 20 maintainers and as a bunch of wrong labels now. This doesnt change any sound, sprites, or ui shit, im just a dumbass.

alphanerdd commented:
Okay so because of some fuckups I made because I suck at github, I'm going to close this PR and reopen it on a fresh PR. I'll link this one in the new version so people can view the discussion.
